### PR TITLE
Next request for server components

### PR DESCRIPTION
**Pull request by agerard-godaddy**
## Summary

Adding a function to get a request-like object unique to the current request in server components.

## Changelog

**@gasket/nextjs**
- Add `request` function for server components

## Test Plan

N/A

### CODE REVIEW

File: packages/gasket-nextjs/src/server/request.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { cookies, headers } from 'next/headers';
+
+const reqCache = new Map();
+
+// TODO: move this to `@gasket/nextjs` package
```
Comment (mmason2-godaddy) [inline packages/gasket-nextjs/src/server/request.js:5]: I think you satisfied the TODO. Nit: doc block

File: packages/gasket-nextjs/test/server/request.test.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+jest.mock('next/headers', () => {
+  return {
+    cookies: mockCookies,
+    headers: mockHeaders
+  };
+});
+
+
+// const request = await import('../../lib/server/request.js').then((module) => module.request);
```
Comment (mmason2-godaddy) [inline packages/gasket-nextjs/test/server/request.test.js:38]: Cleanup

**Review by mmason2-godaddy — APPROVED** [review]

File: packages/gasket-nextjs/src/server/request.js (hunk 1 above)
Comment (agerard-godaddy) [reply packages/gasket-nextjs/src/server/request.js:5]: lolz - I had this in an internal package to start with. Good catch

**Review by jpina1-godaddy — APPROVED** [review]

File: packages/gasket-nextjs/src/server/request.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+const reqCache = new Map();
+
+/**
+ * Get a request-like object unique to the current request in server components.
+ * @param {Object} [query] - Optional query parameters to include in the request
+ * @returns {Request} - Request-like object
+ */
+export function request(query) {
+  const headerStore = headers();
```
[the remark this answers is not in the document]
Comment (agerard-godaddy) [reply packages/gasket-nextjs/src/server/request.js:11]: Yes, it is the same instance per request. I did test this out in a Next app to be certain.

**Pull request merged**